plugins/gpu/nvidia_gpu_

```diff
@@ -87,95 +86,110 @@ if [ "$1" = "suggest" ]; then
 			nGpusCounter=0
-			while [ $nGpusCounter -lt $nGpus ]
+			while [ "$nGpusCounter" -lt "$nGpus" ]
 			do
-				gpuName=`echo "$nGpusOutput" | sed -n $(( $nGpusCounter + 1 ))p | cut -d \( -f 1`
-				echo "temp${nGpusCounter}.warning ${warning:-75}"
-				echo "temp${nGpusCounter}.critical ${critical:-95}"
-				echo "temp${nGpusCounter}.info Temperature information for $gpuName"
-				: $(( nGpusCounter = $nGpusCounter + 1 ))
-			done 
+				gpuName=$(echo "$nGpusOutput" | sed -n $((nGpusCounter+1))p | cut -d \( -f 1)
+				echo "${name}${nGpusCounter}.warning ${warning:-75}"
+				echo "${name}${nGpusCounter}.critical ${critical:-95}"
+				echo "${name}${nGpusCounter}.info Temperature information for $gpuName"
+				: $((nGpusCounter=nGpusCounter+1))
+			done
 			;;
 		mem)
 			# First determine total memory of each GPU...
-			gpusTotalMemOutput=`echo "$smiOutput" | grep -v BAR1 | grep -A 3 "Memory Usage" | grep "Total" | cut -d : -f 2 | tr -d ' '`
+			gpusTotalMemOutput=$(echo "$smiOutput" | grep -v BAR1 | grep -A 3 "Memory Usage" | grep "Total" | cut -d : -f 2 | tr -d ' ')
 			gpusTotalMem=''
 			nGpusCounter=0
-			while [ $nGpusCounter -lt $nGpus ]
+			while [ "$nGpusCounter" -lt "$nGpus" ]
 			do
-				gpuName=`echo "$nGpusOutput" | sed -n $(( $nGpusCounter + 1 ))p | cut -d \( -f 1`
-				echo "mem${nGpusCounter}.info Memory information for $gpuName"
-				gpuMem=`echo "$gpusTotalMemOutput"| sed -n $(( $nGpusCounter + 1 ))p`
+				gpuName=$(echo "$nGpusOutput" | sed -n $((nGpusCounter+1))p | cut -d \( -f 1)
+				echo "${name}${nGpusCounter}.info Memory information for $gpuName"
+				gpuMem=$(echo "$gpusTotalMemOutput"| sed -n $((nGpusCounter+1))p)
 				gpusTotalMem="${gpusTotalMem}${gpuMem} for GPU ${nGpusCounter}"
-				: $(( nGpusCounter = $nGpusCounter + 1 ))
-				if [ $nGpusCounter -lt $nGpus ]; then
+				: $((nGpusCounter=nGpusCounter+1))
+				if [ "$nGpusCounter" -lt "$nGpus" ]; then
 					gpusTotalMem="${gpusTotalMem}, "
 				fi
 			done
 			# ...then output config data.
 			echo 'graph_title GPU memory usage'
 			echo 'graph_args -l 0 -u 100'
-			echo 'graph_vlabel Percentage'
+			echo 'graph_vlabel %'
 			echo 'graph_category memory'
 			echo "graph_info FB Memory usage for NVIDIA GPUs using driver version $driverVersion (total memory is $gpusTotalMem)"
 			;;
 		fan)
 			echo 'graph_title GPU fan speed'
 			echo 'graph_args -l 0 -u 100'
-			echo 'graph_vlabel Percentage'
+			echo 'graph_vlabel %'
 			echo 'graph_category sensors'
 			echo "graph_info Fan speed of NVIDIA GPUs using driver version $driverVersion"
 			nGpusCounter=0
-			while [ $nGpusCounter -lt $nGpus ]
+			while [ "$nGpusCounter" -lt "$nGpus" ]
 			do
-				gpuName=`echo "$nGpusOutput" | sed -n $(( $nGpusCounter + 1 ))p | cut -d \( -f 1`
-				echo "fan${nGpusCounter}.info Fan information for $gpuName"
-				: $(( nGpusCounter = $nGpusCounter + 1 ))
-				done 
+				gpuName=$(echo "$nGpusOutput" | sed -n $((nGpusCounter+1))p | cut -d \( -f 1)
```

Comment:
Please quote.

Maybe you want to move these identical evaluations into a function? (e.g. `get_cpu_name_by_id`)